### DISCUSSION TITLE
Fußbodenheizungsgruppen in Home Assistant verbessern: Level des FALMOT nutzen. (+ Bug?)

### DISCUSSION THREAD

**Discussion by klingbing** [SukramJ/aiohomematic#969] [Ideas]
Zunächst vielen Dank an die  Entwickler für eure Arbeit hier! Ich bin noch neu im Home Assistant Bizznizz, aber "wenn des hier so weiter geht" bin ich zuversichtlich, dass mein Redmatic bald in Rente gehen (ist Redmatic ja schon so, irgendwie) - und durch HA mit Homekit Integration ersetzt wird. DANKE!

Seit einem Firmware Update (vor ... 2 Jahren ...?) des HmIP-FALMOT-C12 Fußbodenheizungsaktors besteht ja die Möglichkeit, mit diesem Heizungsgruppen zu bilden - so weit so gut. Leider wird bis heute noch immer nicht der "Level" (Öffnungsgrad des Ventils am FALMOT) in der Heizungsgruppe abgebildet bzw. mit der Fußbodenheizungsgruppe gesynct, der Level einer Fußbodenheizungsgruppe verharrt bei 0%. Im Homematic-Forum gibt es dazu mehrere Threads.

Da man auf eine Lösung (bzw. Behebung des Problems) von EQ-3 noch lange warten wird (nur so ein Erfahrungswert..), frage ich mich nun, ob nicht vielleicht diese Integration diesbez. eine Verbesserung "herbeizaubern" kann, sodass es zumindest in HA richtig dargestellt wird. Das würde dann neben der richtigen Darstellung in HA z.B. auch dazu führen, dass in Homekit ein "aktives heizen" (orangenfarbene Darstellung der Temperatur) für diese Fußbodenheizungsgruppen angezeigt wird - wie bei einer herkömmlichen Heizungsgruppe mit Heizkörperthermostat - wie es eigentlich sein sollte.

Hier die Gerätedarstellung einer meiner Fußbodenheizungsgruppen, wie zu sehen wird "Level 0,0 %" angezeigt, der dazugehörige Heizkreis des FALMOT ist jedoch aktuell mit 5% geöffnet. 
![grafik](https://user-images.githubusercontent.com/26044205/219877635-f83c7fd4-ea9b-41da-a321-d7a3fe807780.png)

Könnte ich vielleicht als Quick&Dirty Workaround das Gerät irgendwie beeinflussen, sodass sich diese Heizungsgruppe den Level vom tatsächlichen/richtigen Heizkreis des FALMOT abholt statt aus der Heizungsgruppe (welcher ja immer 0% ist)? 

*** *** *** *** ***

Vermutlich ohne Zusammenhang zu dem oben (und vielleicht eher ein Bug report, ich mache das bei Bedarf gern an anderer Stelle auf):

Mir fiel mehr zufällig beim Schreiben hier gerade auf: Eine (einzige) meiner Heizungsgruppen "fehlt" als Gerät. Es handelt sich dabei um eine Fußbodenheizungsgruppe ("Esszimmer Heizungsgruppe"). Diese müsste eigentlich zwischen "Badezimmer.." und "Gäste-WC.." angezeigt werden:

![grafik](https://user-images.githubusercontent.com/26044205/219878480-bea2329a-910a-49dd-b6c4-fff798cef0d4.png)

Das einzige was bei dieser im Gegensatz zu meinen anderen (Fubodenheizungs-)Gruppen anders ist: Sie ist mit **zwei** FALMOT Heizkreisen verknüpft - ansonsten fällt mir nicht ein, was bei dieser anders sein sollte. Es ist auch lediglich das "Gerät" das Fehlt, die Entitäten für die Esszimmer Heizungsgruppe sind vorhanden:

![grafik](https://user-images.githubusercontent.com/26044205/219878350-1dc45edd-1cd6-4646-8f42-114559c1269b.png)

hm.

Ich verwende Raspberrymatic 3.67.10.20230114, falls von Bedeutung

**Comment by SukramJ** [comment]
Da über die API nicht bekannt ist aus welchen Geräten sich eine Gruppe zusammensetzt, kann man auch keine anderen Werte zur Korrektur  "herbeizaubern". Am Besten in der UI statt des `LEVEL` aus der Gruppe, den Sensor aus dem Falmot Gerät verwenden.

**Reply by klingbing** [reply]
OK, schade! Ich war mir nicht sicher wie die Daten in HA ankommen, bzw. wieviel Einfluss man dort dann nehmen kann. Ich hatte gedacht, dass eine Heizungsgruppe vielleicht erst in HA aus den einzelnen Entitäten „zusammengebaut“ wird.

Kannst du deinen Lösungsvorschlag etwas näher erläutern? Ich bin mir nicht sicher an welcher Stelle ich die Level Quelle auswählen kann.

**Reply by SukramJ** [reply]
Was ist da unklar?
Du hast ein Gerät für den Falmot  und dieser hat Sensor Entitäten (ggf. deaktiviert), die den einzelnen Kanälen entsprechen.

**Reply by klingbing** [reply]
Wie ich die einzelne Entität LEVEL des Falmot in der UI anzeigen könnte ist klar. 
Was mir nicht klar ist: Wie ich in der UI ein Thermostat erstellen kann welche sich beim richtigen LEVEL (des FALMOT) bedient. Der Screenshot zeigt links eine herkömmliche Heizungsgruppe, rechts eine Fussbodenheizungsgruppe.

![grafik](https://user-images.githubusercontent.com/26044205/219970716-67ab2ab7-edb8-4eae-a0ce-f1b05b010cf0.png)

Wie bei diesem Beispiel rechts nicht "Leerlauf" sondern "Heizbetrieb" angezeigt wird versuche ich herauszufinden -  auch um dieses Thermostat via Homekit einzubinden um mir dort ein heizen anzeigen zu lassen

**Reply by SukramJ** [reply]
In dem Fall hast du Pech, denn die `climate entity` verwendet ihre eigene `LEVEL` Entität für die Heiz-Aktion.
Wenn ich das richtig sehe kann man für `climate` keine eigenen [Templates](https://www.home-assistant.io/integrations/template/) erstellen.

Vielleicht gibt es in HACS eine Komponente, die  man selber zusammenbauen kann (Better Thermostat?).

**Comment by SukramJ** [comment]
Heist deine "Esszimmer Heizungsgruppe" in HA vielleicht anders? Mal auf ein Entität der "Esszimmer Heizungsgruppe" klicken, und dann unter "Verwandte" das dazugehörige Gerät anzeigen lassen.

**Reply by klingbing** [reply]
Danke! Das war‘s (natürlich), my bad.

**Comment by klingbing** [comment]
Heizsaison eröffnet, Zeit das Thema wieder anzugehen.

> Vielleicht gibt es in HACS eine Komponente, die man selber zusammenbauen kann (Better Thermostat?).

Dieser Hinweis war Gold wert, es gibt tatsächlich eine HACS Lösung! Better Thermostat kann das afaik nicht, aber hiermit https://github.com/jcwillox/hass-template-climate habe ich einen Teilerfolg erzielt beim zusammentemplaten eines eigenen Climate Geräts welche sich dem Level des Falmot bedient - und dadurch die hvac_action (heating oder idle) richtig darstellt!

Hier Code des Test-Geräts in meiner Configuration.yamlm

climate:
  - platform: climate_template
    name: Gaste-WC TEST
    min_temp: 16
    max_temp: 23
    temp_step: 0.5
    current_temperature_template: "{{ states('sensor.gaste_wc_heizungsgruppe_actual_temperature') }}"
    current_humidity_template: "{{ states('sensor.gaste_wc_heizungsgruppe_humidity') }}"
    target_temperature_template: "{{ state_attr('climate.gaste_wc_heizungsgruppe', 'temperature') }}"
    hvac_mode_template: "{{ state_attr('climate.gaste_wc_heizungsgruppe', 'hvac_modes') }}"
    hvac_action_template: "{% set p = states('sensor.technik_ii_fussbodenheizungsaktor_gaste_wc_fussbodenheizungsaktor_heizkreis_level') | float > 0 %} {{  'heating' if p == true else 'idle' }}"

Aber, was ich nicht hinbekomme ist, dass eine manuelle Temperaturänderung/Eingabe des Template Climate Geräts an die HmIP Heizungsgruppe oder das HmIP Wandthermostat übermittelt wird. Dies erfolgt laut Template Climate Doku über „set_temperature“, die Doku velinkt dabei auf die HA Script Doku (https://www.home-assistant.io/docs/scripts) mit diesem Hinweis:
> Defines an action to run when the climate device is given the set temperature command. Can use temperature, target_temp_high, target_temp_low and hvac_mode variables.
Ich komme einfach nicht drauf wie ich hier eine entsprechende action baue.

Ich weiss ich weiss, meine Frage ist vermutlich drüben beim Template Climate besser aufgehoben, ABER die Herausforderung ist doch so HmIP-spezifisch, dass ich die Hoffnung nicht aufgebe, dass sich unter uns hier jemand findet der/die das oben bereits erfolgreich umgesetzt hat - ich drücke mir mal die Daumen! (:
